cloud-controller-manager/do/loadbalancers.go

```diff
@@ -1095,19 +1104,37 @@ func getAlgorithm(service *v1.Service) string {
 	}
 }
 
-func getSizeSlug(service *v1.Service) (string, error) {
-	sizeSlug, ok := service.Annotations[annDOSizeSlug]
+func getSizeSlug(service *v1.Service) string {
+	sizeSlug, _ := service.Annotations[annDOSizeSlug]
+	return sizeSlug
+}
 
-	if !ok || sizeSlug == "" {
-		sizeSlug = "lb-small"
+func getSizeUnit(service *v1.Service) (uint32, error) {
+	sizeUnitStr, ok := service.Annotations[annDOSizeUnit]
+
+	if !ok || sizeUnitStr == "" {
+		sizeUnitStr = "0"
 	}
 
-	switch sizeSlug {
-	case "lb-small", "lb-medium", "lb-large":
-		return sizeSlug, nil
-	default:
-		return "", fmt.Errorf("invalid LB size slug provided: %s", sizeSlug)
+	sizeUnit, err := strconv.Atoi(sizeUnitStr)
+	if err != nil {
+		return 0, fmt.Errorf("invalid LB size unit provided: %s", sizeUnitStr)
 	}
+
+	return uint32(sizeUnit), nil
+}
+
+func validateSizes(sizeSlug string, sizeUnit uint32) error {
+	if sizeSlug != "" && sizeUnit > 0 {
+		return fmt.Errorf("only one of LB size slug and size unit can be provided")
+	} else if sizeSlug != "" {
+		switch sizeSlug {
```

Comment:
Personally, I'd prefer to keep this check/switch in `getSizeSlug` so that the validation stays close to the parsing. Consequently, this function can then focus on validating the combination of the two size parameters (or, alternatively, inline the single `if` statement remaining at the current call site).

Reply:
Moved to the inline approach since there was only one case left